com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl

```diff
@@ -184,6 +184,47 @@
 #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Macros.hlsl"
 #include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Random.hlsl"
 
+
+#if defined(PLATFORM_SUPPORTS_NATIVE_RENDERPASS)
+//entry point
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT(idx, type) UNITY_DECLARE_FRAMEBUFFER_INPUT_##type(idx)
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT_MS(idx, type) UNITY_DECLARE_FRAMEBUFFER_INPUT_##type##_MS(idx)
+// For floats
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT_FLOAT(idx) cbuffer hlslcc_SubpassInput_f_##idx { float4 hlslcc_fbinput_##idx; }
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT_FLOAT_MS(idx) cbuffer hlslcc_SubpassInput_F_##idx { float4 hlslcc_fbinput_##idx[8]; }
+// For halfs
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT_HALF(idx) cbuffer hlslcc_SubpassInput_h_##idx { half4 hlslcc_fbinput_##idx; }
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT_HALF_MS(idx) cbuffer hlslcc_SubpassInput_H_##idx { half4 hlslcc_fbinput_##idx[8]; }
+// For ints
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT_INT(idx) cbuffer hlslcc_SubpassInput_i_##idx { int4 hlslcc_fbinput_##idx; }
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT_INT_MS(idx) cbuffer hlslcc_SubpassInput_I_##idx { int4 hlslcc_fbinput_##idx[8]; }
+// For uints
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT_UINT(idx) cbuffer hlslcc_SubpassInput_u_##idx { uint4 hlslcc_fbinput_##idx; }
+#define UNITY_DECLARE_FRAMEBUFFER_INPUT_UINT_MS(idx) cbuffer hlslcc_SubpassInput_U_##idx { uint4 hlslcc_fbinput_##idx[8]; }
+
+#define UNITY_READ_FRAMEBUFFER_INPUT(idx, v2fname) hlslcc_fbinput_##idx
```

Comment:
Similar, other macros are names `LOAD_TEXTURE2D`. Maybe we should rename this to `LOAD_FRAMEBUFFER_INPUT` or something like that?

Reply:
What about just removing the `UNITY` prefix? as Declare and Read makes most sense in this case, i think

Reply:
`TEXTURE2D(textureName)` declares a Texture2D but it's not named `DECLARE_TEXTURE2D`.
I also don't understand why READ is better than LOAD in this particular case.

Reply:
in my opinion as this involves transient/on-tile data, load makes less sense as we are reading what's already there in the on-tile memory, but i can make it consistent with `TEXTURE2D` and remove Declare and change read to load if you think thats better

Reply:
I don't really be a PITA with this, but since it's public API we basically won't be able to change it later.

I think the `DECLARE` should be removed because it's just inconsistent.
I could see how a `UNITY_` prefix is ok because this input attachment stuff doesn't really map to HLSL directly.

> in my opinion as this involves transient/on-tile data, load makes less sense as we are reading what's already there in the on-tile memory
I don't know, this seems a bit far-fetched to me. In many cases read/load and write/store are used interchangeably. I believe when reading data from memory into registers 'load' is more common. FWIW in GLSL it's called subpassLoad.

I would also remove the DECLARE_FRAMEBUFFER_INPUT that takes the all-caps type as parameter (the one that does the macro concatenation). It's not used in our shaders.

Reply:
ok, so did the naming according to your comment in https://github.com/Unity-Technologies/Graphics/pull/396/commits/95a7bb382f2e16d2735f4cae8000f2c779213f0f